src/search-box/search-box.ts

```diff
@@ -126,7 +124,7 @@ export class NgAisSearchBox extends BaseWidget implements AfterViewInit {
     event.preventDefault();
 
     if (!this.searchAsYouType) {
-      this.state.refine(this.state.query);
+      this.state.refine(this.searchBox.nativeElement.value);
```

Comment:
It looks like the `ref` in React. Do you what is the "standard" way to deal with controlled / uncontrolled input with Angular?

Reply:
Not to my knowledge. It's very reacty.
If you mean having component state flowing from some external model that ReactiveForms come close enough
https://angular.io/guide/forms-overview#data-flow-in-forms